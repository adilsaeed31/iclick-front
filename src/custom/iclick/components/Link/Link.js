/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { Component } from "react";
import PropTypes from "prop-types";
import routes, { Link as NextLink } from "routes";
import track from "lib/tracking/track";
import getConfig from "next/config";

const {
  publicRuntimeConfig: { enableSPARouting }
} = getConfig();

@track((ownProps) => ({
  component: "Link",
  url: ownProps.route || ownProps.href,
  params: ownProps.params
}))
class Link extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    classes: PropTypes.object,
    href: PropTypes.string,
    onClick: PropTypes.func,
    params: PropTypes.object,
    route: PropTypes.string,
    to: PropTypes.string
  }

  render() {
    const {
      children,
      classes,
      className,
      href,
      onClick,
      params,
      route,
      tracking, // eslint-disable-line
      to,
      ...props
    } = this.props;

    if (enableSPARouting === false) {
      const {
        urls: { as }
      } = routes.findAndGetUrls(route || to || href, params);

      return (
        <a className={className} href={as}>
          {children}
        </a>
      );
    }

    return (
      <NextLink route={route || to || href} params={params} {...props} passHref>
        <a className={className} role="link" tabIndex={0}>
          {children}
        </a>
      </NextLink>
    );
  }
}

export default Link;
