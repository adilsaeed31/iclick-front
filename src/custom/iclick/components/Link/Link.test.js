import React from "react";
import renderer from "react-test-renderer";
import Link from "./";

test("basic snapshot", () => {
  const component = renderer.create(<Link route="/next">Linked</Link>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
