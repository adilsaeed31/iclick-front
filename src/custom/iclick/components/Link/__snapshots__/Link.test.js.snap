// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<a
  className="SkLink-anchor-1"
  href="/next"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
>
  Linked
</a>
`;
